File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/go_solutions/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.go
func minFlips(grid [][]int) int {
    n := len(grid)
    res := 0
    for i := 0; i < n/2; i++ {
        for j := 0; j < n/2; j++ {
            cnt := grid[i][j] + grid[i][n-1-j] + grid[n-1-i][j] + grid[n-1-i][n-1-j]
            res += min(cnt, 4-cnt)
        }
    }
    if n%2 == 1 {
        for i := 0; i < n/2; i++ {
            cnt := grid[n/2][i] + grid[n/2][n-1-i]
            res += min(cnt, 2-cnt)
        }
    }
    return res
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.rs
impl Solution {
    pub fn min_flips(grid: Vec<Vec<i32>>) -> i32 {
        let n = grid.len();
        let mut res = std::i32::MAX;

        for mask in 0..(1 << n) {
            let mut flips = 0;
            let mut valid = true;

            for i in 0..n {
                for j in 0..n {
                    let expected = ((mask >> i) & 1) ^ ((mask >> j) & 1);
                    if grid[i][j] != expected {
                        flips += 1;
                    }
                }
            }

            if valid {
                res = res.min(flips);
            }
        }

        res
    }
}